src/NWAClient.ts

```diff
@@ -0,0 +1,249 @@
+      relayUrl: options.relayUrl,
+      appSecretKey,
+      appPubkey: getPublicKey(hexToBytes(appSecretKey)),
+      name: options.name,
+      icon: options.icon,
+      returnTo: options.returnTo,
+      requestMethods: options.requestMethods,
+      notificationTypes: options.notificationTypes,
+      maxAmount: options.maxAmount,
+      expiresAt: options.expiresAt,
+      budgetRenewal: options.budgetRenewal,
+      isolated: options.isolated,
+      metadata: options.metadata,
+    };
+
+    if (!this.options.relayUrl) {
+      throw new Error("Missing relay url");
+    }
+    if (!this.options.requestMethods) {
+      throw new Error("Missing request methods");
+    }
+    this.relay = new Relay(this.options.relayUrl);
+
+    if (globalThis.WebSocket === undefined) {
+      console.error(
+        "WebSocket is undefined. Make sure to `import websocket-polyfill` for nodejs environments",
+      );
+    }
+  }
+
+  /**
+   * returns the NWA connection URI which should be given to the wallet
+   */
+  get connectionUri() {
+    return this.getConnectionUri();
+  }
+
+  /**
+   * returns the NWA connection URI which should be given to the wallet
+   * @param nwaSchemeSuffix open a specific wallet. e.g. "alby" will set the scheme to
+   * nostr+walletauth+alby to ensure the link will be opened in an Alby wallet
+   */
+  getConnectionUri(nwaSchemeSuffix = "") {
+    const searchParams = new URLSearchParams({
+      relay: this.options.relayUrl,
+      request_methods: this.options.requestMethods.join(" "),
+      ...(this.options.name ? { name: this.options.name } : {}),
+      ...(this.options.icon ? { icon: this.options.icon } : {}),
+      ...(this.options.returnTo ? { return_to: this.options.returnTo } : {}),
+      ...(this.options.notificationTypes
+        ? {
+            notification_types: this.options.notificationTypes.join(" "),
+          }
+        : {}),
+      ...(this.options.maxAmount
+        ? { max_amount: this.options.maxAmount.toString() }
+        : {}),
+      ...(this.options.budgetRenewal
+        ? { budget_renewal: this.options.budgetRenewal }
+        : {}),
+      ...(this.options.expiresAt
+        ? { expires_at: this.options.expiresAt.toString() }
+        : {}),
+      ...(this.options.isolated
+        ? { isolated: this.options.isolated.toString() }
+        : {}),
+      ...(this.options.metadata
+        ? { metadata: JSON.stringify(this.options.metadata) }
+        : {}),
+    });
+
+    return `nostr+walletauth${nwaSchemeSuffix ? `+${nwaSchemeSuffix}` : ""}://${this.options.appPubkey}?${searchParams
+      .toString()
+      .replace(/\+/g, "%20")}`;
+  }
+
+  static parseWalletAuthUrl(walletAuthUrl: string): NWAOptions {
+    // makes it possible to parse with URL in the different environments (browser/node/...)
+    // parses with or without "//"
+    walletAuthUrl = walletAuthUrl
+      .replace("nostr+walletconnect://", "http://")
+      .replace("nostr+walletconnect:", "http://");
```

Comment:
Shouldn't these be `nostr+walletauth`? 🤔

Reply:
And should we also remove nostr+walletauth`+suffix` before doing this replace?

Reply:
Good catch, thanks